src/Tools/Auditor/Report.dfy

```diff
@@ -0,0 +1,125 @@
+include "../../Utils/Library.dfy"
+
+module AuditReport {
+  import opened Utils.Lib.Seq
+
+  //// Data types for report ////
+
+  datatype Tag =
+    | IsGhost
+    | IsSubsetType
+    | IsCallable
+    | HasNoBody // TODO: should all tags be positive?
+    | HasAxiomAttribute
+    | HasExternAttribute
+    | HasVerifyFalseAttribute
+    | HasAssumeInBody // Maybe: (expr: Expression) TODO: :axiom?
+    | HasRequiresClause
+    | HasEnsuresClause
+    | HasNoWitness // TODO: should all tags be positive?
+    | HasJustification
+    // TODO: decreases *?
```

Comment:
I think that would be nice to have as well.

Reply:
I agree, though it'll require translating `WitnessExpr`. The decision on what we want to do with that in the semantics in the short term is the biggest open question for me.

Reply:
Based on our discussion, we don't need to treat is as an expression: we can use `Option` or similar types everywhere `*` is used.

Reply:
Let's update the comment correspondingly

Reply:
A tag now exists for this, but it's never created. Issue #17 tracks extracting the information needed to create it.